docs/content/docs/getting-started/use-built-provider.md

```diff
@@ -74,8 +74,12 @@ To stop using the local provider binary, you can run:
 rm -rf ~/.terraform.d/plugins/registry.terraform.io/hashicorp/
 ```
 
+## Download production providers
+
+When in this mode, Terraform will be unable to pull providers from the registry using `terraform init`, even if a version constraint requires them. However, providers downloaded to the discovery directory may be used by `init`. To download providers to the discovery directory, run `terraform providers mirror ~/.terraform.d/plugins` from your configuration's directory prior to running `terraform init`.
```

Comment:
I haven't had any issues using this workflow without `terraform providers mirror` - that is, Terraform seems to be perfectly happy to download the `random` provider (and others) from the registry even if I'm using the local provider. Are you sure this is required? Is this a restriction of older Terraform versions?

Reply:
Ah, other providers are fine. Just the google ones are affected.

```
$ terraform init

Initializing the backend...

Initializing provider plugins...
- Finding hashicorp/google-beta versions matching "~> 4.41.0"...
╷
│ Warning: Version constraints inside provider configuration blocks are deprecated
│
│   on main.tf line 2, in provider "google-beta":
│    2:   version = "~> 4.41.0"
│
│ Terraform 0.13 and earlier allowed provider version constraints inside the provider configuration block, but that is now deprecated and will be removed in a future version of Terraform. To
│ silence this warning, move the provider version constraint into the required_providers block.
╵

╷
│ Error: Failed to query available provider packages
│
│ Could not retrieve the list of available versions for provider hashicorp/google-beta: no available releases match the given constraints ~> 4.41.0
```